### ISSUE TITLE
All CPU Meter and Drives Meter are crashing Rainmeter since today.

### ISSUE THREAD

**Issue by DarkNecrotic** [SilverAzide/Gadgets#44]
I just can't find what's going on. Everything worked as intended yesterday. I am on the newest version of Rainmeter (4.5.13.3632 64-bit) and HWInfo (7.22-4731) and Windows 10 with all current updates. I think there were some updates yesterday (namely KB5011831 and KB5012157). Today rainmeter suddenly won't show up on start. So I, one by one, deactivated every skin and tried starting it up. If I deactivate All CPU Meter and Drives Meter in the Rainmeter.ini, rainmeter starts as normal and all other skins are working. I also got GPU Meter, Network Meter, Clock and Weather Meter of your suite aktive and they are working fine. Only CPU and Drives are not working anylonger. Every time i activate them, rainmeter crashes without an error-message.

**Comment by DarkNecrotic** [comment]
I just found the same issue with Top Process Meter. All 4 variants are not working anymore (CPU, GPU, IO and RAM).
And I also found out that some of the additional gadgets from DudeBaker have the same issue. If this is something on my End, i will continue looking, but I can't say for sure if a windows update can introduce problems to rainmeter skins.

**Comment by Tintin-Asterix** [comment]
Same thing for me. These ones work:
Battery, Calender, Weather, Network Meter, Welcome, Clocks. GPU.

These ones make it instantly crash:
All CPU Meter, Drives Meter, Top Process Meter.

So it seems to be ones that are more focused on pulling info from the system. I tried turning/uninstalling/downgrading/etc HWiNFO and deleting any config files etc but it still crashes.

I can't help but feel it might be something else though, I installed an older version of Gadgets like 7.1.0 and it still crashes. Did something in Windows change?

**Comment by DarkNecrotic** [comment]
One of those updates (KB5012157) was an update for .NET Framework 3.5 and 4.8. Maybe there is the problem? I don't code, so it's just a guess.

**Comment by miquelfire** [comment]
Those are preview updates, so anyone who doesn't manually check for updates won't get them. If there is something in those updates that is causing the crash, everyone should start crashing when whatever broke is pushed out to everyone automatically next week on patch Tuesday.

**Comment by DarkNecrotic** [comment]
That's true... let's be patient then.
It's just very strange... and my 3rd monitor looks so naked now :)

**Comment by Tintin-Asterix** [comment]
> Those are preview updates, so anyone who doesn't manually check for updates won't get them. If there is something in those updates that is causing the crash, everyone should start crashing when whatever broke is pushed out to everyone automatically next week on patch Tuesday.

I'm on Windows LTSC 2022 and don't have any preview updates. The last one I got was KB5012117.

**Comment by DarkNecrotic** [comment]
KB5012117 installed nearly one month ago on my system.

**Comment by SilverAzide** [comment]
Hello Everyone!  Sorry you are having issues!
I am not seeing this crashing issue (yet??).  However, one thing that seems to be common with what you are all reporting is that the skins that cause a crash are all ones that use the UsageMonitor plugin.  Skins like the Clock, Weather Meter, etc. don't use this plugin.

Please try the procedure described here:  **[HOW TO: Fix your Performance Counters database](https://github.com/SilverAzide/Gadgets/wiki/HOW-TO-Fix-your-Performance-Counters-database#how-to-fix-your-performance-counters-database)**.  Let me know if this helps!

**Comment by Valus3000** [comment]
I am having the issue also on Win10 Pro 21H1 x64. 
The CPU Meter just not moving at all, everything just says 0%. Only start happening today.

Latest update was :  2022-04 Cumulative Update Preview for Windows 10 Version 21H1 for x64-based Systems (KB5011831)

**Comment by SilverAzide** [comment]
> **Edit:** Doing the procedure you describe above, fix my issue. Thank you

Excellent news!  Thank you for the feedback!

**Comment by DarkNecrotic** [comment]
> Hello Everyone! Sorry you are having issues!
> 
> I am not seeing this crashing issue (yet??). However, one thing that seems to be common with what you are all reporting is that the skins that cause a crash are all ones that use the UsageMonitor plugin. Skins like the Clock, Weather Meter, etc. don't use this plugin.
> 
> Please try the procedure described here: **[HOW TO: Fix your Performance Counters database](https://github.com/SilverAzide/Gadgets/wiki/HOW-TO-Fix-your-Performance-Counters-database#how-to-fix-your-performance-counters-database)**. Let me know if this helps!
> 
> **Edit:** I just installed the following Windows Updates, rebooted, and so far (fingers crossed) I am not having any issues: 2022-04 Cumulative Update Preview for .NET Framework 3.5 and 4.8 for Windows 10 Version 21H2 for x64 (KB5012157) 2022-04 Cumulative Update Preview for Windows 10 Version 21H2 for x64-based Systems (KB5011831)
> 
> Rainmeter 4.5.13.3632 (64-bit) Language: English (1033) Build time: 2022-03-23 15:23:40 Windows 10 Pro 2009 64-bit (build 19044) - English (1033)


Always ask the guy who build it!! Cause he knows best :) THANK YOU SO MUCH... your trick with the performance counter database did it for me. I checked my perfmon and found out, that there are no more counters to choose. So entered your magic lines into windows powershell and tadaaaa.... everything is working again.
I just had to try it two times, because the first time i forgot to run powershell in admin mode. ;)

But one question stays... what does cause this to happen? Very interesting...

**Comment by SilverAzide** [comment]
I have no idea why this happens.  I know I can cause it to happen (as I mention in my info article), but when I do it, it doesn't cause crashing.  I actually had this happen to me about a month ago TWICE in about a 15 minute interval on a Windows 11 machine, and I have no idea what I did.  Perhaps Windows did an update in the background.

Anyway, glad it is working for you!  Thanks for reporting! I posted a new thread about this on the Rainmeter forums, as many people seem to be getting this crashing issue all of a sudden.
